extension/spark-sink/src/main/scala/org/apache/spark/sql/palo/PaloDfsLoadTask.scala

```diff
@@ -0,0 +1,219 @@
+
+  private val dataDir = checkAndInit(PaloConfig.PALO_DATA_DIR) // hdfs or afs
+  private val loadCmd = checkAndInit(PaloConfig.LOADCMD)
+  private val ugi = checkAndInit("hadoop.job.ugi")
+  private val broker = checkAndInit(PaloConfig.BROKER_NAME)
+
+  private[palo] val paloTimeout = conf.getInt(PaloConfig.PALO_TIME_OUT, 0)
+  private[palo] val maxFilterRatio = conf.getDouble(PaloConfig.MAX_FILTER_RATIO, 0)
+  private[palo] val loadDeletFlag = conf.get(PaloConfig.LOAD_DELETE_FLAG, "false")
+  private[palo] val is_negative = conf.get(PaloConfig.IS_NEGATIVE, "false").toBoolean
+  assert(maxFilterRatio >= 0 && maxFilterRatio <= 1 && paloTimeout >= 0)
+
+  private[palo] var fs: FileSystem = null
+  private[palo] var dataPath: Path = null
+  private[palo] var outputStream: FSDataOutputStream = null
+  private[palo] var writer: BufferedWriter = null
+
+  // init outputStream
+  init()
+
+  /**
+   * check the key which must be specified
+   */
+  private def checkAndInit(key: String): String = {
+    val value = conf.get(key)
+    if (value == null) {
+      throw new IllegalArgumentException(s"${key} must be specified!!")
+    }
+    value
+  }
+
+  private def init() {
+    try {
+      fs = FileSystem.newInstance(conf)
+      // dataPath is the path of tmp file to store data genererated by Dataset
+      // dataPath = dataDir + label
+      //  --dataDir specified by customer
+      //  --label generated by PaloWriterTask
+      dataPath = new Path(dataDir, label)
+      logInfo(s"dataDir:${dataDir}, label:${label}, dataPath:${dataPath}")
+
+      // dataPath may exists for failover, delete the dataPath and create new one firstly
+      if (fs.exists(dataPath)) {
+        fs.delete(dataPath, true)
+        logInfo(s"dataPath:${dataPath} exists, delete it successfully")
+      }
+      outputStream = fs.create(dataPath)
+      writer = new BufferedWriter(new OutputStreamWriter(outputStream, "UTF-8"))
+
+      logInfo(s"init successfully, create a new dataPath:${dataPath} to store contents")
+    } catch {
+      case e: IOException =>
+        logError(s"PaloDfsLoadTask init failed: ${e.getMessage}")
+        try {
+          if (outputStream != null) {
+            outputStream.close
+          }
+        } catch {
+          case e1: IOException =>
+          logError(s"close outputStream error when init failed:${e1.getMessage}")
+        }
+        try {
+          if (writer != null) {
+            writer.close
+          }
+        } catch {
+           case e1: IOException =>
+           logError(s"close BufferedWriter error when init failed:${e1.getMessage}")
+        }
+        throw e
+    }
+  }
+
+  override def close(): Unit = {
+    super.close
+    try {
+      if (fs != null) {
+        fs.close
+      }
+    } catch {
+      case e: IOException => logError(s"close failed for ${e.getMessage}")
+      throw e
+    }
+  }
+
+  /**
+   * TODO: need to check "\n" whether in the end of content?
```

Comment:
Doris does not require a "\n" at the end of load source file.
So is this TODO for Doris, or for some other reason?